dataeng/resources/warehouse-transforms.sh

```diff
@@ -65,6 +72,51 @@ fi
 dbt $DBT_COMMAND $FULL_REFRESH_ARG --models $MODEL_SELECTOR $exclude_param --profile $DBT_PROFILE --target $DBT_TARGET --profiles-dir $WORKSPACE/analytics-secure/warehouse-transforms/ ; ret=$?;
 postCommandChecks "run" $ret ;
 
+# The setup here is being done in order to integrate monte carlo and dbt
+# This feature was requested as part of DESUPPORT-1358
+
+# set up token path
+export VAULT_TOKEN_PATH=${WORKSPACE}/vault-config/vault-token
+# set path for config file which store token in each job workspace
+export VAULT_CONFIG_PATH=${WORKSPACE}/vault-config/vault_config
+
+# write credentials to vault server to get the token
+vault write -field=token auth/approle/login \
+    role_id=${ANALYTICS_VAULT_ROLE_ID} \
+    secret_id=${ANALYTICS_VAULT_SECRET_ID} \
+| vault login -no-print token=-
+
+# set vault token
+
+# Creating separate token for each job in its workspace
+# By default token is generated in home directory of server
+# When token location is changed using VAULT_CONFIG_PATH vault cli
+# should find the new location of token using the vault config file
+# This is the expected behaviour for vault cli. But vault cli was
+# not working as expected and is not able to locate the new token location
+# Have to explicitly store token in token environment variable so that
+# vault cli can use the newly generated token for each job
+
+export VAULT_TOKEN="$(cat ${WORKSPACE}/vault-config/vault-token)"
+
+# set monte carlo api keys to integrate with monte carlo
+for KEY in ID TOKEN; do
+
+    export MCD_DEFAULT_API_${KEY}="$(vault kv get -version=1 -field=MCD_DEFAULT_API_${KEY} kv/monte-carlo-api-keys)"
```

Comment:
We already have the monte carlo API key in third_part_apis path in the vault, let's use that.